servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -489,7 +494,10 @@ public long create(TachyonURI path, CreateOptions options)
 
     mTTLBuckets.insert(inode);
 
-    MasterContext.getMasterSource().incFilesCreated(created.size());
+    MasterContext.getMasterSource()
+        .incFilesCreated(Iterables.size(Iterables.filter(created, InodeFile.class)));
+    MasterContext.getMasterSource()
+        .incDirectoriesCreated(Iterables.size(Iterables.filter(created, InodeDirectory.class)));
```

Comment:
I don't think you have to iterate over the list twice. Since this code path, we know we are creating an inode, all created inodes in the path will be directories, except the last one (which will be a file). Therefore, you could just add 1 to FileCreated, and the "size() - 1" to DirectoriesCreated.